contracts/modules/claims/Incidents.sol

```diff
@@ -0,0 +1,234 @@
+    underlyingTokens[coveredToken] = underlyingToken;
+  }
+
+  function addIncident(
+    address coveredToken,
+    uint incidentDate,
+    uint priceBefore,
+    uint priceAfter
+  ) external onlyGovernance {
+
+    require(priceBefore > priceAfter, "Incidents: No depeg");
+
+    address underlying = underlyingTokens[coveredToken];
+    require(underlying != address(0), "Incidents: Unsupported token");
+
+    Incident memory incident = Incident(
+      uint32(incidentDate),
+      uint112(priceBefore),
+      uint112(priceAfter),
+      coveredToken
+    );
+    incidents.push(incident);
+  }
+
+  function getCoverDetails(QuotationData qd, uint coverId) internal view returns (
+    address coveredToken,
+    address coverOwner,
+    uint coverStartDate,
+    uint coverExpirationDate,
+    uint sumAssured,
+    bytes4 currency
+  ) {
+
+    (
+    /* id */, coverOwner, coveredToken,
+    currency, sumAssured, /* premiumNXM */
+    ) = qd.getCoverDetailsByCoverID1(coverId);
+
+    uint coverPeriod = uint(qd.getCoverPeriod(coverId)).mul(1 days);
+    coverExpirationDate = qd.getValidityOfCover(coverId);
+    coverStartDate = coverExpirationDate.sub(coverPeriod);
+  }
+
+  function redeemPayout(
+    uint coverId,
+    uint incidentId,
+    uint coveredTokenAmount
+  ) external returns (uint claimId, uint payoutAmount) {
+
+    Incident memory incident = incidents[incidentId];
+    address coverOwner;
+    address coveredToken;
+    address underlyingToken;
+    uint sumAssured;
+
+    {
+      QuotationData qd = quotationData();
+      uint coverExpirationDate;
+      uint coverStartDate;
+      bytes4 currency;
+
+      (
+      coveredToken, coverOwner, coverStartDate,
+      coverExpirationDate, sumAssured, currency
+      ) = getCoverDetails(qd, coverId);
+
+      // check ownership & validity
+      require(msg.sender == coverOwner, "Incidents: Not cover owner");
+      require(coverStartDate >= incident.date, "Incidents: Cover start date before the incident");
+      require(coverExpirationDate <= incident.date, "Incidents: Cover end date after the incident");
+
+      {
+        // check covered protocol
+        require(coveredToken == incident.coveredToken, "Incidents: Covered token != incident token");
+
+        // note: technically this check should have happened at cover purchase time
+        underlyingToken = underlyingTokens[coveredToken];
+        address coverCurrencyAddress = claimsReward().getCurrencyAssetAddress(currency);
+        require(coverCurrencyAddress == underlyingToken, "Incidents: Cover asset != underlying asset");
+      }
+
+      // decrese total sum assured
+      qd.subFromTotalSumAssured(currency, sumAssured);
+      qd.subFromTotalSumAssuredSC(coveredToken, currency, sumAssured);
+    }
+
+    {
+      // min/max checks. sumAssured assumes 18 decimals
+      uint coverAmount = sumAssured.mul(1e18);
+      uint maxAmount = coverAmount.div(incident.priceBefore);
```

Comment:
if you have sumAssured in wad (units) you need to multiply by 1e36
because you then divide by price which is in wei so that maxAmount stays in wei